### PR TITLE
feat(shopping-cart): Add multiple payment-gateway to shopping-cart

### PR DESCRIPTION
**Pull request by prafful1234**
Problem -
- There could only be one default gateway for checkout.
- The shopping settings cart had no options to add multiple gateways for checkout. 

Changes-
- add gateway selector item table.
- add gateway selector to shopping cart checkout.
- linked default payment gateway to the default in the payment gateway table.
- add a patch to updated old default payment gateway to payment gateway table as default in shopping cart settings.
- auto-fetch gateway and label on gateway_account selection.

- Shopping Cart settings > Gateways

Screenshot from 2020-04-22 18-55-37

- Shopping cart checkout.

Screenshot from 2020-04-22 19-00-23

- Standard Payment Flow on selected gateway.

Screenshot from 2020-04-22 19-00-28

### CODE REVIEW

**Comment by ossplus** [comment]
https://github.com/frappe/webshop/issues/343

**Review by DeveloperMetal — DISMISSED** [review]

**Review by DeveloperMetal — COMMENTED** [review]
Does this PR align with the new cart direction?

File: erpnext/shopping_cart/doctype/gateway_selector_item/gateway_selector_item.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{
```
Comment (rohitwaghchaure) [inline erpnext/shopping_cart/doctype/gateway_selector_item/gateway_selector_item.json]: "Gateway Selector Item" is not a good name of the doctype, maybe use "Shopping Cart Payment Gateway"

Comment (prafful1234) [reply erpnext/shopping_cart/doctype/gateway_selector_item/gateway_selector_item.json]: made required changes, please give it a check. thanks.

**Comment by Alchez** [comment]
@rohitwaghchaure The tests are seem to be failing for unrelated logic (fetching fiscal years), can you please have a look?

**Comment by Alchez** [comment]
@rohitwaghchaure, can you please re-open this one?

**Pull request closed without merging**